doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,487 @@
+We want our program to output a list of nodes that are currently reachable based on the information in the relation :code:`Links`.
+Thus, let's add an output relation :code:`ConnectedNodes` and provide rules how to calculate records in this relation *(mind the presence of the dot at the end of each rule)*.
+To do that we are going to add few more lines to the :code:`t1_reachability_monitor.dl` so it will look as follows:
+
+.. code-block::
+
+    input relation Links(src: string, dst: string, link_status: bool)
+    output relation ConnectedNodes(src: string, dst: string)
+
+    ConnectedNodes(src, dst) :- Links(src, dst, true).
+    ConnectedNodes(src, dst) :- ConnectedNodes(src, intermediate_node),
+                                Links(intermediate_node, dst, true).
+
+Let's briefly discuss the rule to calculate :code:`ConnectedNodes` relation.
+First, we say that any two directly connected nodes such that the :code:`link_status` between them is :code:`true` belong to the :code:`ConnectedNodes` relation.
+Furthermore, any two nodes are considered connected if there is path from the source node to the destination node such that all directly connected nodes along the path also have the :code:`link_status` between them to be :code:`true`.
+Take a minute here to realize how easy it is to write such rule in DDlog.
+While these rules may not require much effort in other programming languages, the distinct feature of DDlog is that the computing is done incrementally!
+Isn't is amazing?
+
+With that done, let's compile and start our program.
+We will feed our DDlog code to the DDlog compiler and it will generate a new folder.
+This folder contains CLI so we can run our program as well as a library that we will need later for our C program.
+Note, that on the first run it will take a while, but will be signifanctly faster afterwards.
+After the code is compiled let's start a command-line so we can start playing with our program.
+
+.. code-block::
+
+    $ ddlog -i t1_reachability_monitor.dl
+    $ cd t1_reachability_monitor_ddlog/ && cargo build --release
+    $ ./target/release/t1_reachability_monitor_cli
+
+If we display the contents of the :code:`ConnectedNodes` relation we will see that this it is empty, which make sense as we haven't populated :code:`Links` relation with anything yet:
+
+.. code-block::
+
+    >> dump ConnectedNodes;
+
+To exit DDlog shell press `[CTRL+d]`.
+If you are using `Windows Terminal` do not press `[CTRL+D]` (note capital `D`, i.e., `[CTRL+SHIFT+d]`) as it will open a new tab.
+
+Let's supply some input data.
+We are going to insert several records into :code:`Links` relation.
+But first let's define a simple network topology that we are going to work through.
+To make our life more interesting (and just slightly more complicated) the links between nodes will not be bidirectional by default.
+
+.. image:: ./images/topology.jpg
+
+Note that DDlog is a `transaction-based`_ programming language.

```

Comment:
You should probably be more precise here: there are really no transactions in the *.dl file, but when you supply inputs you do it using transactions; the outputs are also associated to transactions.